### ISSUE TITLE
Horizontal scroll bar missing in HTML for wide class="dotgraph" objects

### ISSUE THREAD

**Issue by btokarev** [doxygen/doxygen#8924]
While building the documentation pages with wide images (autogenerated to appear as part of the div class="dotgraph"), the side scroll bar does not appear as it used to in version 1.8.13.

**Expected behavior**
Side scroll bar should appear as expected for dotgraph class objects.

**Screenshots**
![изображение](https://user-images.githubusercontent.com/95291022/144062176-8a759203-3666-439c-aee0-9e2a93cae1db.png)

**To Reproduce**
It is expected this will reproduce when generating the documentation page with the wide image if it is included to div class="dotgraph".

**Version**
Version 1.8.13 displays the scroll bar correctly. 
![изображение](https://user-images.githubusercontent.com/95291022/144064077-f22d995c-67d6-40a2-a2de-a645b6824f3d.png) 
[1.8.13.zip](https://github.com/doxygen/doxygen/files/7626448/1.8.13.zip)

Versions 1.9.1 and 1.9.2 does not display the scrollbar as at the first screenshot.
[1.9.1.zip](https://github.com/doxygen/doxygen/files/7626436/1.9.1.zip)
[1.9.2.zip](https://github.com/doxygen/doxygen/files/7626455/1.9.2.zip)

Scrollbar appears as expected for 1.9.2 for images that are part of div class="image" 
![изображение](https://user-images.githubusercontent.com/95291022/144065168-48480c07-c8f8-448e-8737-2ff1ba60a92d.png)
[longpage.zip](https://github.com/doxygen/doxygen/files/7626493/longpage.zip)

**Comment by albert-github** [comment]
I see the problem you are facing, it looks like to have to do with an overflow in some `<li>` components.
In your example page I see:

![image](https://user-images.githubusercontent.com/5223533/144092762-67900332-49e7-436e-8e3f-5cd7f461b8e5.png)

and the strange thing is that the image is part of the last list item.
In the resulting html file I see:
```
<div class="textblock"><ul>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qabugs.html">Package QABugs</a> </li>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qadnaming.html">Package QADNaming</a> </li>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qadraw.html">Package QADraw</a> </li>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qancollection.html">Package QANCollection</a> <div class="dotgraph">
<img src="Open%20CASCADE%20Technology%20Toolkit%20TKQADraw_files/dot_schema_TKQADraw.png" alt="dot_schema_TKQADraw.png" usemap="#dot_schema_TKQADraw.map" border="0">
<map name="dot_schema_TKQADraw.map" id="dot_schema_TKQADraw.map"><area shape="rect" href="file:///C:/build/doc/refman/html/toolkit_tkqadraw.html" title="TKQADraw" alt="" coords="2437,5,2552,53">
<area shape="rect" href="file:///C:/build/doc/refman/html/toolkit_tkbrep.html" title="TKBRep" alt="" coords="5,101,96,149">
...
<area shape="rect" href="file:///C:/build/doc/refman/html/toolkit_tkstdl.html" title="TKStdL" alt="" coords="4901,101,4985,149">
</map>
</div>
 </li>
</ul>
</div></div><!-- contents -->
```
whilst I would expect:
```
<div class="textblock"><ul>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qabugs.html">Package QABugs</a> </li>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qadnaming.html">Package QADNaming</a> </li>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qadraw.html">Package QADraw</a> </li>
<li><a class="el" href="file:///C:/build/doc/refman/html/package_qancollection.html">Package QANCollection</a>
 </li>
</ul>
</div>
 <div class="dotgraph">
<img src="Open%20CASCADE%20Technology%20Toolkit%20TKQADraw_files/dot_schema_TKQADraw.png" alt="dot_schema_TKQADraw.png" usemap="#dot_schema_TKQADraw.map" border="0">
<map name="dot_schema_TKQADraw.map" id="dot_schema_TKQADraw.map"><area shape="rect" href="file:///C:/build/doc/refman/html/toolkit_tkqadraw.html" title="TKQADraw" alt="" coords="2437,5,2552,53">
<area shape="rect" href="file:///C:/build/doc/refman/html/toolkit_tkbrep.html" title="TKBRep" alt="" coords="5,101,96,149">
...
<area shape="rect" href="file:///C:/build/doc/refman/html/toolkit_tkstdl.html" title="TKStdL" alt="" coords="4901,101,4985,149">
</map>
</div>
</div><!-- contents -->
```

Note the different place of the:
```
 </li>
</ul>
</div>
```

To be able to find the source of this problem. I found the https://dev.opencascade.org/doc/occt-7.6.0/refman/html/index.html and https://dev.opencascade.org/doc/occt-7.6.0/refman/html/toolkit_tkqadraw.html (build with doxygen 1.9.1)  the later showing the problem as well. So some extra information is needed:
- where does the bullet list part come from?


I assume you used the `https://git.dev.opencascade.org/repos/occt.git` (I already cloned it with: `git clone https://git.dev.opencascade.org/repos/occt.git occt`) but it is not clear how to build the documentation (and how long it will take).
- Please specify in command line terms how to build the documentation without using non standard tools or having to build the complete product first. Best would be to have a `Doxyfile` so I can just start `doxygen Doxyfile`.

I might have found the source of the problem: `adm/occaux.tcl` it looks like here something is done with some extra pages and it might be that here the lists are not closed properly.

**Comment by btokarev** [comment]
Thank you for looking into this!

The suggested position for closing tags indeed fixes the problem immediately; however, I am still a little bit curious about why is it working correctly in the version 1.8.13. Tags are at the same position there as in resulting file from 1.9.1 (list is closed improperly), but the scroll bar remains intact. You can see this page live at https://dev.opencascade.org/doc/occt-7.5.0/refman/html/toolkit_tkqadraw.html
![изображение](https://user-images.githubusercontent.com/95291022/144224602-23f73497-6896-4830-a402-44cb357ca8f1.png)
The overlapping part is seen on the screenshot, but for 1.8.13 it keeps the scrollbar on it's rightful place.

As for the building system, the bullet list is generated by Graphviz I suppose (did not really dug deep into this part yet).

To build the documentation in HTML it seems to be 2 third-party tools needed (the full procedure is described here https://dev.opencascade.org/doc/overview/html/occt_contribution__documentation.html):
Tcl/Tk Version 8.5 or 8.6: https://www.tcl.tk/software/tcltk/download.html
Dot Part of Graphviz software, used by Doxygen for generation of class diagrams in Reference Manual: https://www.graphviz.org/download/

After that, the generation starts at gendoc by calling 
`gendoc.bat -refman -html -s=server` for the Reference manual, part of which is the issue-subject page.

This will generate a doxyfile based on the requested data and will proceed to doxygen. `adm/occaux.tcl` seems to be a part of pre-building steps; I will try to trace these steps.

**Comment by albert-github** [comment]
I didn't investigate properly the cause in the css file, but I saw that for the `<li>` tag something was done in respect to the `overflow` handling (between doxygen 1.8.14 and 1.8.15).

Did some investigations and it looks like the root cause is #636. Some correction shave already been made with #754, but it looks like you stumbled here on another one.

Still I think the real problem here lies in the improper way the `<li>` etc. tags are  closed and this is now shown by means of the given image.

**Comment by albert-github** [comment]
I just created a small test case showing the problem: [example.tar.gz](https://github.com/doxygen/doxygen/files/7633611/example.tar.gz)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8928

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
